x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/ApiKeyGenerator.java

```diff
@@ -57,22 +54,18 @@ public void generateApiKey(Authentication authentication, CreateApiKeyRequest re
             apiKeyService.createApiKey(authentication, request, roleDescriptors, listener);
         }, listener::onFailure);
 
-        if (ServiceAccountSettings.REALM_NAME.equals(authentication.getSourceRealm().getName())) {
-            final ServiceAccount serviceAccount = ServiceAccountService.getServiceAccounts().get(authentication.getUser().principal());
-            if (serviceAccount == null) {
-                roleDescriptorsListener.onFailure(
-                    new ElasticsearchSecurityException(
-                        "the authentication is created by a service account that does not exist: ["
-                            + authentication.getUser().principal()
-                            + "]"
-                    )
-                );
-            } else {
-                roleDescriptorsListener.onResponse(Set.of(serviceAccount.roleDescriptor()));
-            }
-        } else {
-            rolesStore.getRoleDescriptors(new HashSet<>(Arrays.asList(authentication.getUser().roles())), roleDescriptorsListener);
+        final Subject effectiveSubject = AuthenticationContext.fromAuthentication(authentication).getEffectiveSubject();
+
+        // TODO: retain current behaviour that User of an API key authentication has no roles.
```

Comment:
Can you clarify this TODO?
Is it something to fix in this PR, or should someone fix it in the future?

Reply:
It is just a comment, not TODO ... I will remove the `TODO:` part.